doc/src/modules/polys/numberfields.rst

```diff
@@ -0,0 +1,289 @@
+following list, based on [Cohen93]_ (Sec. 4.9.3).
+
+For a number field $K = \mathbb{Q}(\theta)$, whose ring of algebraic integers
+is denoted $\mathbb{Z}_K$, compute:
+
+1. an integral basis of $\mathbb{Z}_K$
+2. the decomposition of rational primes in $\mathbb{Z}_K$
+3. $\mathfrak{p}$-adic valuations for ideals and elements
+4. the Galois group of the Galois closure of $K$
+5. a system of fundamental units of $K$
+6. the regulator $R(K)$
+7. the class number
+8. the structure of the class group $Cl(K)$
+9. decide whether a given ideal is principal, and if so compute a generator.
+
+As a foundation, and to support our basic ability to define and work with
+number fields and algebraic numbers, we also set the following problems:
+
+10. Given an algebraic number -- expressed by radicals and rational operations,
+    or even as a special value of a transcendental function -- determine its
+    minimal polynomial over $\mathbb{Q}$.
+11. Given an irreducible polynomial, distinguish its roots as elements of
+    $\mathbb{C}$ by computing isolating intervals.
+12. The Primitive Element Problem: Given several algebraic numbers
+    $\alpha_1, \ldots, \alpha_m$, compute a single algebraic number $\theta$
+    such that $\mathbb{Q}(\alpha_1, \ldots, \alpha_m) = \mathbb{Q}(\theta)$.
+13. The Field Isomorphism Problem: Decide whether two number fields
+    $\mathbb{Q}(\alpha)$, $\mathbb{Q}(\beta)$ are isomorphic.
+14. The Field Membership Problem: Given two algebraic numbers $\alpha$,
+    $\beta$, decide whether $\alpha \in \mathbb{Q}(\beta)$, and if so write
+    $\alpha = f(\beta)$ for some $f(x) \in \mathbb{Q}[x]$.
+
+At present only a subset of the tasks enumerated in the lists above is yet
+supported in SymPy, and if you are interested in expanding support, you are
+encouraged to contribute!
+
+At time of writing, the existing solutions to the above problems are found
+in the following places:
+
+=================================  ======================================
+Task                               Implementation
+=================================  ======================================
+(1) integral basis                 :py:func:`~.round_two`
+(2) prime decomposition            :py:func:`~.prime_decomp`
+(3) $\mathfrak{p}$-adic valuation  :py:func:`~.prime_valuation`
+(10) find min poly                 :py:func:`~.minpoly`
+(11) isolating intervals           :py:func:`~.isolate`
+(12) primitive element             :py:func:`~.primitive_element`
+(13) field isomorphism             :py:func:`~.field_isomorphism`
+(14) field membership              :py:func:`~.to_number_field`
+=================================  ======================================
+
+
+Internals
+=========
+
+Algebraic number fields
+-----------------------
+
+Algebraic number fields are represented in SymPy by the
+:py:class:`~.AlgebraicField` class, which is a part of the
+polynomial domains system.
```

Comment:
I think it would be good to add a
```
:ref:`polys-domainsref`
```
here which links through to here:
https://docs.sympy.org/latest/modules/polys/domainsref.html#reference-docs-for-the-poly-domains

Reply:
Ah, good. I didn't know how to make links of this kind (i.e. using `:ref:`).